components/inputmethod/ibus/patches/01-enable-autostart-management.patch

```diff
@@ -0,0 +1,81 @@
++++ ibus-1.5.13/setup/setup.ui	2017-10-26 20:06:46.964623585 +0000
+@@ -1033,7 +1033,6 @@
+               <object class="GtkBox" id="vbox4">
+                 <property name="orientation">vertical</property>
+                 <property name="can_focus">False</property>
+-                <property name="no_show_all">True</property>
+                 <child>
+                   <placeholder/>
+                 </child>
+@@ -1060,7 +1059,6 @@
+                 <child>
+                   <object class="GtkFrame" id="frame1">
+                     <property name="can_focus">False</property>
+-                    <property name="no_show_all">True</property>
+                     <property name="label_xalign">0</property>
+                     <property name="shadow_type">none</property>
+                     <child>
+--- ibus-1.5.13/setup/main.py.~1~	2016-02-22 11:46:23.000000000 +0000
++++ ibus-1.5.13/setup/main.py	2017-10-26 20:36:10.684124059 +0000
+@@ -25,6 +25,7 @@
+ 
+ import os
+ import signal
++import subprocess
+ import sys
+ import time
+ 
+@@ -506,35 +507,26 @@
+         model.set(iter, COLUMN_PRELOAD, data[DATA_PRELOAD])
+ 
+     def __is_auto_start(self):
+-        link_file = path.join(GLib.get_user_config_dir(),
+-                              "autostart/ibus.desktop")
+-        ibus_desktop = path.join(os.getenv("IBUS_PREFIX"),
+-                                 "share/applications/ibus.desktop")
+-
+-        if not path.exists(link_file):
+-            return False
+-        if not path.islink(link_file):
+-            return False
+-        if path.realpath(link_file) != ibus_desktop:
+-            return False
+-        return True
+-
+-    def __checkbutton_auto_start_toggled_cb(self, button):
+-        auto_start_dir = path.join(GLib.get_user_config_dir(), "autostart")
+-        if not path.isdir(auto_start_dir):
+-            os.makedirs(auto_start_dir)
+-
+-        link_file = path.join(GLib.get_user_config_dir(),
+-                              "autostart/ibus.desktop")
+-        ibus_desktop = path.join(os.getenv("IBUS_PREFIX"),
+-                                 "share/applications/ibus.desktop")
+-        # unlink file
++        home = os.getenv("HOME")
++        config_file = path.join(home, ".imf")
+         try:
+-            os.unlink(link_file)
++            if os.path.isfile(config_file):
++	        im = subprocess.check_output(["/usr/bin/sh", config_file, "-name"])
++                if im.startswith("IBus "):
++                    return True
+         except:
+             pass
+-        if self.__checkbutton_auto_start.get_active():
+-            os.symlink(ibus_desktop, link_file)
++        return False
++
++    def __checkbutton_auto_start_toggled_cb(self, button):
++        IBUS_SCRIPT="/etc/X11/xinit/imf.d/0010.ibus"
++
++        home = os.getenv("HOME")
++        if home:
++            if self.__checkbutton_auto_start.get_active():
++               os.system("/usr/bin/cat %s > %s/.imf" % (IBUS_SCRIPT, home))
```

Comment:
Why os.system and not subprocess.call or subprocess.check_call?

Reply:
What's the actual difference?  We need to execute shell command. I'd worried more about path escaping here, perhaps, this exact call could be rewritten in clear python.

Reply:
Avoided using os.system here. Also new code shouldn't care about file path escaping